dsa-solutions/gfg-solutions/Easy problems/bfs-of-graph.md

```diff
@@ -0,0 +1,222 @@
+        
+        while (!queue.isEmpty()) {
+            int node = queue.poll();
+            traversal.add(node);
+            for (int neighbor : adj.get(node)) {
+                if (!visited[neighbor]) {
+                    queue.add(neighbor);
+                    visited[neighbor] = true;
+                }
+            }
+        }
+        return traversal;
+    }
+}
+```
+
+  </TabItem>
+  <TabItem value="cpp" label="C++">
+
+```cpp
+#include <vector>
+#include <queue>
+
+class Solution {
+public:
+    std::vector<int> bfsOfGraph(int V, std::vector<int> adj[]) {
+        std::vector<int> result;
+        std::vector<bool> visited(V, false);
+        std::queue<int> q;
+        
+        q.push(0);
+        visited[0] = true;
+        
+        while (!q.empty()) {
+            int node = q.front();
+            q.pop();
+            result.push_back(node);
+            for (int neighbor : adj[node]) {
+                if (!visited[neighbor]) {
+                    q.push(neighbor);
+                    visited[neighbor] = true;
+                }
+            }
+        }
+        return result;
+    }
+};
+```
+
+  </TabItem>
+  <TabItem value="javascript" label="JavaScript">
+
+```javascript
+class Solution {
+    bfsOfGraph(V, adj) {
+        let visited = new Array(V).fill(false);
+        let queue = [0];
+        visited[0] = true;
+        let result = [];
+        
+        while (queue.length > 0) {
+            let node = queue.shift();
+            result.push(node);
+            for (let neighbor of adj[node]) {
+                if (!visited[neighbor]) {
+                    queue.push(neighbor);
+                    visited[neighbor] = true;
+                }
+            }
+        }
+        return result;
+    }
+}
+```
+
+  </TabItem>
+  <TabItem value="typescript" label="TypeScript">
+
+```typescript
+class Solution {
+    bfsOfGraph(V: number, adj: number[][]): number[] {
+        let visited = new Array(V).fill(false);
+        let queue: number[] = [0];
+        visited[0] = true;
+        let result: number[] = [];
+        
+        while (queue.length > 0) {
+            let node = queue.shift() as number;
+            result.push(node);
+            for (let neighbor of adj[node]) {
+                if (!visited[neighbor]) {
+                    queue.push(neighbor);
+                    visited[neighbor] = true;
+                }
+            }
+        }
+        return result;
+    }
+}
+```
+
+  </TabItem>
+</Tabs>
+
+---
+
+## Complexity analysis
+
+The provided solutions efficiently perform a Breadth First Search (BFS) traversal of a directed graph. By starting from the 0th vertex and using a queue to manage the traversal, the algorithms ensure that all reachable vertices are visited in the correct order. The solutions operate in O(V + E) time and use O(V) space complexity, where V and E are the numbers of vertices and edges in the graph, respectively.
+
+**Time Complexity:** $ O(V + E)  $
```

Comment:
not using space

and replace `$ O(V + E) $` to `$O(V + E)$`

and also, `$ O(V)$` to `$O(V)$`